Mathmagician/Program.cs

```diff
@@ -3,33 +3,80 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Mathmagician.Extras;
 
 namespace Mathmagician
 {
     class Program
     {
         static void Main(string[] args)
         {
+            //Introduce Math Wizard.
+            Console.WriteLine(ASCII.Wizard());
+            Console.ForegroundColor = ConsoleColor.DarkMagenta;
+            Console.WriteLine("Greetings. I am the math wizard. Give me the ingredients and I will concoct a splendid list of numbers for you.");
+            Console.WriteLine("If you would like to quit at any time, simply tell me to 'shut up'.");
+            Console.ResetColor();
+
+
+            //List of possible commands for users.
+            List<string> commands = new List<string>() { "Integers", "Primes", "Fibonacci", "even", "odd" };
+
+            //Repeats until user quits.
             bool running = true;
             while (running)
             {
-                List<string> commands = new List<string>() { "Integers", "Primes", "Fibonacci", "even", "odd" };
+                
+                //Ask user what they would like to do, then possible commands.
                 Console.WriteLine("What would you like for me to do?");
                 for (int i = 0; i < commands.Count; i++)
                 {
-                    Console.WriteLine(commands[i]);
+                    Console.WriteLine(commands[i] );
                 }
+
+                //Prompt user for command input.
                 Console.Write("> ");
                 string commandInput = Console.ReadLine();
-                Console.WriteLine("How many should I print?");
-                Console.Write("> ");
-                string listAmountInput = Console.ReadLine();
-                int listAmountParsed = Int32.Parse(listAmountInput);
+                if (commandInput.ToLower() == "shut up")
+                {
+                    Environment.Exit(0);
+                }
 
-                Console.WriteLine("Cool, I'm going to print {0} {1} numbers.", listAmountInput, commandInput);
+                //Prompt user for iteration input.
+                bool responseUnverified = true;
+                string listAmountInput = "";
+                int listAmountParsed = 0;
+                while (responseUnverified)
+                {
+                    Console.WriteLine("How many should I print?");
+                    Console.Write("> ");
+                    listAmountInput = Console.ReadLine();
 
+                    //Checks to see if user entered valid number, if they instruct to quit, then exit loop and close program, if not, then throw error and repeat instruction.
+                    if (Int32.TryParse(listAmountInput, out listAmountParsed))
+                    {
+                        responseUnverified = false;
+                    }
+                    else
+                    {
+                        if (listAmountInput.ToLower() == "shut up")

```

Comment:
You make this check twice. In a future refactor, you might want to extract this checking functionality into method.